cli/cmd/repair.go

```diff
@@ -43,6 +51,27 @@ linkerd upgrade.`,
 				return errors.New("secret/linkerd-config-overrides already exists. If you need to regenerate this resource, please delete it before proceeding")
 			}
 
+			// Check if the CLI version matches with that of the server
+			clientVersion := version.Version
+			var serverVersion string
+			publicClient, err := public.NewExternalClient(cmd.Context(), controlPlaneNamespace, k8sAPI)
+			if err != nil {
+				return err
+			}
+			serverVersion, err = healthcheck.GetServerVersion(cmd.Context(), publicClient)
+			if err != nil {
+				return err
+			}
+
+			if serverVersion != clientVersion {
+				helperVersion := serverVersion
+				// Use 2.9.4 CLI version for all 2.9 versions
+				if versionRegex.Match([]byte(serverVersion)) {
+					helperVersion = "stable-2.9.4"
+				}
+				return fmt.Errorf("Please run the repair command with a CLI that has the same version as the control plane.\nRun `LINKERD2_VERSION=\"%s\"; curl -sL https://run.linkerd.io/install | sh` to install the server version of the CLI", helperVersion)
```

Comment:
TIOLI: May I suggest doing instead `curl -sL https://run.linkerd.io/install | LINKERD2_VERSION=\"%s\" sh` to avoid leaking that env var into the current environment?

Reply:
OTOH, if `helperVersion` falls back to `stable-2.9.4` with say a server at `stable-2.8.1`, the message will be confusing ("has the same version"...)

Reply:
The command I suggested works for me in Linux, but might not in other systems. If so, I think it's fine to leave it as you initially had it, and not complicating things by unsetting the var at the end.